### DISCUSSION TITLE
Execution of queries doesn't returns error when context Done is closed in execution process?

### DISCUSSION THREAD

**Discussion by stasvo** [jackc/pgx#1299] [Q&A]
Do I understand correct that execution of queries doesn't returns error if context Done is closed in execution process? 
I see that for example if call cancel while the request in process of being executed then the execution process stops but the returned error is nil.

**Comment by jackc** [comment]
No, you should get a `context.Canceled` or `context.DeadlineExceeded`.

**Reply by stasvo** [reply]
For example in my test I always get nil error when use Query function and then call CancelFunc of context .
For test I use query with very long execution time.
```
rows, err := pool.Query(ctx, `select column from very_huge_table where something;`)
fmt.Println("should be error:", err)
```
What is see in terminal is: should be error: nil

So in this case i get nil error.

**Reply by jackc** [reply]
See https://pkg.go.dev/github.com/jackc/pgx/v4#Conn.Query

> Err() on the returned Rows must be checked after the Rows is closed to determine if the query executed successfully as some errors can only be detected by reading the entire response. e.g. A divide by zero error on the last row.

**Reply by stasvo** [reply]
Thanks for answer.
